Add Hercules DJControl Inpulse 500 mapping

resetreboot opened the pull request:
[Manual PR](https://github.com/mixxxdj/manual/pull/725)

resetreboot committed:
Add Hercules DJControl Inpulse 500 mapping

JoergAtGithub commented:
Welcome at Mixxx!
As a first-time contributor we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+  0xFF8800: 0x74,
+});
+
+// Constants
+
+DJCi500.EFFECT_ONLY_MODE = 1;
+DJCi500.FILTER_AND_EFFECT_MODE = 2;
```
```suggestion
// Constants
DJCi500.EFFECT_ONLY_MODE = 1;
DJCi500.FILTER_AND_EFFECT_MODE = 2;
```

cr7pt0gr4ph7 replied:
```suggestion
// Constants
DJCi500.EFFECT_ONLY_MODE = 1;
DJCi500.FILTER_AND_EFFECT_MODE = 2;
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+DJCi500.slicerBeatsPassed = [0, 0, 0, 0];
+DJCi500.slicerPreviousBeatsPassed = [0, 0, 0, 0];
+DJCi500.slicerTimer = [false, false, false, false];
+//DJCi500.slicerJumping = [0, 0, 0, 0];
```
`slicerJumping` is not used anywhere else in this file.
```suggestion
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+//  * Add beat jump + SHIFT jumps
+//
+// ****************************************************************************
+var DJCi500 = {};
+///////////////////////////////////////////////////////////////
+//                       USER OPTIONS                        //
+///////////////////////////////////////////////////////////////
```
```suggestion
var DJCi500 = {};

///////////////////////////////////////////////////////////////
//                       USER OPTIONS                        //
///////////////////////////////////////////////////////////////
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+///////////////////////////////////////////////////////////////
+//                          SLICER                           //
+///////////////////////////////////////////////////////////////
+DJCi500.selectedSlicerDomain = [8, 8, 8, 8]; //length of the Slicer domain
+//PioneerDDJSX.slicerDomains = [8, 16, 32, 64];
```
The second line seems to be a copy-paste leftover.

```suggestion
DJCi500.selectedSlicerDomain = [8, 8, 8, 8]; // Length of the Slicer domain
//PioneerDDJSX.slicerDomains = [8, 16, 32, 64];
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+const pairColorsOn = [0x1F, 0x1F, 0x03, 0x03, 0x74, 0x74, 0x60, 0x60];
+const pairColorsOff = [0x12, 0x12, 0x02, 0x02, 0x4C, 0x4C, 0x40, 0x40];
+
+
```
Small formatting nitpick: IMO only one blank line should be used here for consistency.

```suggestion
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+  this.vinylButtonState = [DJCi500.initialVinylMode, DJCi500.initialVinylMode, DJCi500.initialVinylMode, DJCi500.initialVinylMode];
+
+  // Pitch ranges and status
+  this.pitchRanges = [0.08, 0.10, 0.15, 0.16, 0.24, 0.50, 0.90]; //select pitch range
+  this.pitchRangeId = 0; //id of the array, one for each deck
```
```suggestion
  this.pitchRanges = [0.08, 0.10, 0.15, 0.16, 0.24, 0.50, 0.90]; // Select pitch range
  this.pitchRangeId = 0; // id of the array, one for each deck
```

cr7pt0gr4ph7 reviewed (changes requested):
A few small formatting nitpicks.

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+      this.input = function (channel, control, value, status, group) {
+        if (value === 0x7F) {
+          if (engine.getValue(deckData.currentDeck, "play_latched")) {      //play_indicator play_latched
+            var deck = parseInt(deckData.currentDeck.charAt(8));
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
            var deck = script.deckFromGroup(deckData.currentDeck);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+    unshift: function () {
+      this.input = function(channel, _control, value, status, group) {
+        if (value === 0x7F) {
+          var deck = parseInt(deckData.currentDeck.charAt(8)) - 1;
+          var new_status = !deckData.vinylButtonState[deck];
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
          var deck = script.deckFromGroup(deckData.currentDeck);
          var new_status = !deckData.vinylButtonState[deck - 1];
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+    shift: function () {
+      this.input = function (channel, control, value, status, group) {
+        if (value === 0x7F){
+          var deck = parseInt(deckData.currentDeck.charAt(8)) - 1;
+          deckData.slowPauseSetState[deck] = !deckData.slowPauseSetState[deck];
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
          var deck = script.deckFromGroup(deckData.currentDeck);
          deckData.slowPauseSetState[deck - 1] = !deckData.slowPauseSetState[deck - 1];

```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+    rpm: 33 + 1/3,
+    group: "[Channel"+midiChannel+"]",
+    inputWheel: function(_channel, _control, value, _status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
      var deck = script.deckFromGroup(deckData.currentDeck);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+      }
+    },
+    inputTouch: function(channel, control, value, status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
      var deck = script.deckFromGroup(deckData.currentDeck);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+    rpm: 33 + 1/3,
+    group: "[Channel"+midiChannel+"]",
+    inputWheel: function(_channel, _control, value, _status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
      var deck = script.deckFromGroup(deckData.currentDeck);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+      }
+    },
+    inputTouch: function(channel, control, value, status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
      var deck = script.deckFromGroup(deckData.currentDeck);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+  for (var i = 1; i <= 8; i++) {
+    this.padSelectButtons[i] = new components.Button({
+      midi: [0x90 + midiChannel, 0x0F + (i - 1)],
+      input: function(channel, control, value, status, group) {
+        var deck = parseInt(deckData.currentDeck.charAt(8)) - 1;
+        if (control === 0x11) {
+          DJCi500.slicerActive[deck] = true;
+        } else {
+          DJCi500.slicerActive[deck] = false;
+        }
+      },
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions) and prefix unused parameters with `_`.
```suggestion
      input: function(_channel, control, _value, _status, _group) {
        var deck = script.deckFromGroup(deckData.currentDeck);
        if (control === 0x11) {
          DJCi500.slicerActive[deck - 1] = true;
        } else {
          DJCi500.slicerActive[deck - 1] = false;
        }
      },
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+}
+
+DJCi500.spinback_button = function(channel, control, value, status, group) {
+  var deck = parseInt(group.substring(8,9)); // work out which deck we are using
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
    var deck = script.deckFromGroup(group);      
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+  var deckA = DJCi500.deckA.currentDeck;
+  var deckB = DJCi500.deckB.currentDeck;
+
+  //Tempo:
```
Usual comment style would be to have a space after the `//`. This applies to all the other places below, too.
```suggestion
  // Tempo:
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+    if ((tempo2 / tempo1) > 1.5){doubleTempo = 1; diff = tempo1 - tempo2/2;}
+  }
+
+  if ( diff < -0.25)
```
Formatting nitpick: Consistent whitespace should be used.
```suggestion
  if (diff < -0.25)
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+
+  //Check double tempo:
+  var doubleTempo = 0;
+  if (diff > 0){
+    if ((tempo1 / tempo2) > 1.5){doubleTempo = 1; diff = tempo1/2 - tempo2;}
+  }
+  else{
+    if ((tempo2 / tempo1) > 1.5){doubleTempo = 1; diff = tempo1 - tempo2/2;}
+  }
```
Formatting nitpick: Consistent formatting of braces.
```suggestion
  if (diff > 0){
    if ((tempo1 / tempo2) > 1.5) {
      doubleTempo = 1;
      diff = tempo1 / 2 - tempo2;
    }
  } else {
    if ((tempo2 / tempo1) > 1.5) {
      doubleTempo = 1;
      diff = tempo1 - tempo2 / 2;
    }
  }
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+
+    //Do beat alignement only if the tracks are already on Tempo
+    // and only if they are playing
+    if ( (engine.getValue(deckA, "play_latched")) && (engine.getValue(deckB, "play_latched")) ){
```
Formatting nitpick: Consistent whitespace should be used, unnecessary parentheses can be omitted.
```suggestion
    if (engine.getValue(deckA, "play_latched") && engine.getValue(deckB, "play_latched")){

```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+
+  // Update the vinyl button
+  var vinylState = false;
+  var deckIndex = parseInt(group.charAt(8)) - 1;
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
  var deckIndex = script.deckFromGroup(deckData.currentDeck) - 1;
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+//this below is connected to beat_active
+DJCi500.slicerBeatActive = function(value, group, control) {
+  // This slicer implementation will work for constant beatgrids only!
+  var deck = parseInt(group.charAt(8)) - 1;
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions) and rename `deck` to `deckIndex` when its 0-based instead of 1-based.
```suggestion
  var deck = script.deckFromGroup(deckData.currentDeck) - 1;
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+///////////////////////////////////////////////////////////////
+DJCi500.slicerButtonFunc = function(channel, control, value, status, group) {
+  var index = control - 0x20,
+    deck = parseInt(group.charAt(8)) - 1,
```
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions) and rename `deck` to `deckIndex` when its 0-based instead of 1-based.
```suggestion
    deck = script.deckFromGroup(deckData.currentDeck) - 1,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1566 @@
+              var bpm_file = engine.getValue(group, "file_bpm"),
+                playposition = engine.getValue(group, "playposition"),
+                duration = engine.getValue(group, "duration");
+              /*
+              if (Math.floor((playposition * duration) * (bpm_file / 60.0)) > endLoop) {
+                engine.setValue(group, "beatjump", -8);
+              }*/
```
Can the commented out code be removed completely?
```suggestion
```

cr7pt0gr4ph7 reviewed (changes requested):
A few actual code change suggestions as well as some more formatting nitpicks.

cr7pt0gr4ph7 commented on res/controllers/Hercules_DJControl_Inpulse_500.midi.xml:
```diff
@@ -0,0 +1,3424 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
_(Placing this comment at the top, though this applies to the whole file)._

The controller mapping XML files should be formatted with an indentation of 4 spaces (though some existing files deviate from this and are indented with 2 spaces instead).

This file currently contains a mix of tabs and spaces as well as varying indentation.

cr7pt0gr4ph7 reviewed (changes requested):
I can't actually test this mapping as I don't have the associated hardware, though this seems to look good so far. The indentation should be fixed to only use spaces, though.

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
Update res/controllers/Hercules-DJControl-Inpulse-500-script.js

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot committed:
fix: re-tabbed to 4 spaces

JoergAtGithub commented:
The pre-commit check is failing. The best way to fix pre-commit issues is to install pre-commit locally on your system, as described here: https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking
Than it will fix these issues automatically before every commit.
Alternatively, you can download the pre-commit.patch file from the artifacts of this PR
![grafik](https://github.com/mixxxdj/mixxx/assets/64457745/c3076931-5c8e-4389-b464-f6b4622140b3)
unzip it, and apply it using:
`git apply pre-commit.patch`

resetreboot committed:
Fix pre-commit

m0dB commented:
Hi @resetreboot , the linter is flagging up quite a large amount of issues in the .js code. If you want I can help out fixing those, but I guess we should coordinate this, so we don't step on each others toes.

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+    let active = false;
+
+    if (group === "[EffectRack1_EffectUnit1]") {
+        active = engine.getValue(group, `group_${  deckA  }_enable`);
```
```suggestion
        active = engine.getValue(group, `group_${deckA}_enable`);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        } else {
+            midi.sendShortMsg(0x96, 0x63, 0x00);
+        }
+        active = engine.getValue(group, `group_${  deckB  }_enable`);
```
```suggestion
        active = engine.getValue(group, `group_${deckB}_enable`);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        } else {
+            midi.sendShortMsg(0x96, 0x67, 0x00);
+        }
+        active = engine.getValue(group, `group_${  deckB  }_enable`);
```
```suggestion
        active = engine.getValue(group, `group_${deckB}_enable`);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            midi.sendShortMsg(0x97, 0x63, 0x00);
+        }
+    } else if (group === "[EffectRack1_EffectUnit2]") {
+        active = engine.getValue(group, `group_${  deckA  }_enable`);
```
```suggestion
        active = engine.getValue(group, `group_${deckA}_enable`);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+
+    if (group == `[QuickEffectRack1_${  deckA  }]`) {
+        midi.sendShortMsg(0x96, 0x66, active ? 0x1C : 0x60);
+    } else if (group == `[QuickEffectRack1_${  deckB  }]`) {
```
```suggestion
    } else if (group == `[QuickEffectRack1_${deckB}]`) {
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+    const deckB = DJCi500.deckB.currentDeck;
+    const active = engine.getValue(group, "enabled");
+
+    if (group == `[QuickEffectRack1_${  deckA  }]`) {
```
```suggestion
    if (group == `[QuickEffectRack1_${deckA}]`) {
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+    for (let k = 1; k <= 3; k++) {
+        this.eqKnob[k] = new components.Pot({
+            midi: [0xB0 + midiChannel, 0x01 + k],
+            group: `[EqualizerRack1_${  this.currentDeck  }_Effect1]`,
```
```suggestion
            group: `[EqualizerRack1_${this.currentDeck}_Effect1]`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        this.eqKnob[k] = new components.Pot({
+            midi: [0xB0 + midiChannel, 0x01 + k],
+            group: `[EqualizerRack1_${  this.currentDeck  }_Effect1]`,
+            inKey: `parameter${  k}`,
```
```suggestion
            inKey: `parameter${k}`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            sendShifted: false,
+            on: 0x5C,
+            off: 0x30,
+            outKey: `beatloop_${  fractions[i - 1]  }_enabled`,
```
```suggestion
            outKey: `beatloop_${fractions[i - 1]}_enabled`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            on: 0x5C,
+            off: 0x30,
+            outKey: `beatloop_${  fractions[i - 1]  }_enabled`,
+            inKey: `beatloop_${  fractions[i - 1]  }_toggle`,
```
```suggestion
            inKey: `beatloop_${fractions[i - 1]}_toggle`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            sendShifted: false,
+            on: 0x5C,
+            off: 0x30,
+            outKey: `beatloop_${  shiftFractions[i - 1]  }_enabled`,
```
```suggestion
            outKey: `beatloop_${shiftFractions[i - 1]}_enabled`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            on: 0x5C,
+            off: 0x30,
+            outKey: `beatloop_${  shiftFractions[i - 1]  }_enabled`,
+            inKey: `beatloop_${  shiftFractions[i - 1]  }_toggle`,
```
```suggestion
            inKey: `beatloop_${shiftFractions[i - 1]}_toggle`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            sendShifted: true,
+            on: 0x1F,
+            off: 0x12,
+            key: `beatlooproll_${  fractions[i - 1]  }_activate`,
```
```suggestion
            key: `beatlooproll_${fractions[i - 1]}_activate`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            shiftOffset: 8,
+            shiftControl: true,
+            sendShifted: true,
+            group: `[EffectRack1_EffectUnit${  midiChannel  }_Effect${  i  }]`,
```
```suggestion
            group: `[EffectRack1_EffectUnit${midiChannel}_Effect${i}]`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        shiftOffset: 8,
+        shiftControl: true,
+        sendShifted: true,
+        group: `[QuickEffectRack1_[Channel${  midiChannel  }]]`,
```
```suggestion
        group: `[QuickEffectRack1_[Channel${midiChannel}]]`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        shiftOffset: 8,
+        shiftControl: true,
+        sendShifted: true,
+        group: `[QuickEffectRack1_[Channel${  midiChannel  }]]`,
```
```suggestion
        group: `[QuickEffectRack1_[Channel${midiChannel}]]`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        shiftOffset: 8,
+        shiftControl: true,
+        sendShifted: true,
+        group: `[QuickEffectRack1_[Channel${  midiChannel  }]]`,
```
```suggestion
        group: `[QuickEffectRack1_[Channel${midiChannel}]]`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        input: function(channel, _control, value, _status, group) {
+            if (value === 0x7F) {
+                const deckGroup = deckData.currentDeck;
+                script.toggleControl(this.group, `group_${  deckGroup  }_enable`);
```
```suggestion
                script.toggleControl(this.group, `group_${deckGroup}_enable`);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        input: function(_channel, _control, value, _status, _group) {
+            if (value === 0x7F) {
+                const deckGroup = deckData.currentDeck;
+                script.toggleControl(this.group, `group_${  deckGroup  }_enable`);
```
```suggestion
                script.toggleControl(this.group, `group_${deckGroup}_enable`);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+                engine.setValue(`[EffectRack1_EffectUnit${  this.number  }]`, "super1", Math.abs(script.absoluteNonLin(value, 0.0, 0.5, 1.0, 0, 127) - 0.5)*2);
+            } else {
+                // Move the filter knob
+                engine.setValue(`[QuickEffectRack1_${  deckData.currentDeck  }]`, "super1", script.absoluteNonLin(value, 0.0, 0.5, 1.0, 0, 127));
```
```suggestion
                engine.setValue(`[QuickEffectRack1_${deckData.currentDeck}]`, "super1", script.absoluteNonLin(value, 0.0, 0.5, 1.0, 0, 127));
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            sendShifted: true,
+            on: pairColorsOn[i - 1],
+            off: pairColorsOff[i - 1],
+            jump: `beatjump_${  jmpVal  }${movement}`,
```
```suggestion
            jump: `beatjump_${jmpVal}${movement}`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+            on: pairColorsOn[i - 1],
+            off: pairColorsOff[i - 1],
+            jump: `beatjump_${  jmpVal  }${movement}`,
+            jumpShift: `beatjump_${  jmpValShft  }${movement}`,
```
```suggestion
            jumpShift: `beatjump_${jmpValShft}${movement}`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+        input: function(channel, control, value, status, group) {
+            if (DJCi500.updateEffectStatus(midiChannel, deckData.currentDeck)) {
+                // Move the effects knobs
+                engine.setValue(`[EffectRack1_EffectUnit${  this.number  }]`, "super1", Math.abs(script.absoluteNonLin(value, 0.0, 0.5, 1.0, 0, 127) - 0.5)*2);
```
```suggestion
                engine.setValue(`[EffectRack1_EffectUnit${this.number}]`, "super1", Math.abs(script.absoluteNonLin(value, 0.0, 0.5, 1.0, 0, 127) - 0.5)*2);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+    this.filterKnob = new components.Pot({
+        midi: [0xB0 + midiChannel, 0x01],
+        number: midiChannel,
+        group: `[QuickEffectRack1_[Channel${  midiChannel  }]]`,
```
```suggestion
        group: `[QuickEffectRack1_[Channel${midiChannel}]]`,
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+    DJCi500.fxSelIndicator(0, "[EffectRack1_EffectUnit1]", 0, 0);
+    DJCi500.fxSelIndicator(0, "[EffectRack1_EffectUnit2]", 0, 0);
+
+    DJCi500.fxEnabledIndicator(0, `[QuickEffectRack1_${  group  }]`, 0, 0);
```
```suggestion
    DJCi500.fxEnabledIndicator(0, `[QuickEffectRack1_${group}]`, 0, 0);
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+DJCi500.updateEffectStatus = function(midiChannel, channel) {
+    let status = false;
+    for (let i = 1; i <= 3; i++) {
+        status = status || engine.getValue(`[EffectRack1_EffectUnit${  midiChannel  }_Effect${  i  }]`, "enabled");
```
```suggestion
        status = status || engine.getValue(`[EffectRack1_EffectUnit${midiChannel}_Effect${i}]`, "enabled");
```

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1537 @@
+    const deck = script.deckFromGroup(deckData.currentDeck) - 1;
+    const channel = deck % 2;
+
+    print(`***** SLICER ACTIVE VALUE: ${  DJCi500.slicerActive[deck]}`);
+    print(`***** SLICER: deck ${  deck  } channel ${  channel}`);
```
```suggestion
    print(`***** SLICER ACTIVE VALUE: ${DJCi500.slicerActive[deck]}`);
    print(`***** SLICER: deck ${deck} channel ${channel}`);
```

cr7pt0gr4ph7 reviewed (changes requested):
Another round of formatting nitpicks that should probably be fixed, mainly unusual formatting of template strings. I would heavily suggest using an autoformatter to format the code.

**Tip:** Instead of committing all the suggestions individually, you can use [**Add suggestion to batch**](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#:~:text=Add%20suggestion%20to%20batch) to commit multiple proposed fixes at once.

resetreboot committed:
Apply suggestions from code review

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

resetreboot commented:
Hey @m0dB I would totally love getting some help. I applied the patch from `pre-commit` but still it didn't like it. I had it enabled, but I guess the new batch of commits need a pass to check that the linter is happy again.

JoergAtGithub commented:
> I applied the patch from `pre-commit` but still it didn't like it. I had it enabled, but I guess the new batch of commits need a pass to check that the linter is happy again.

@resetreboot The patch file contains only automatic fixes for formatting issues. Correct English spelling (spellcheck) and syntax errors (eslint) are only detected, but need to be fixed by the developer themself.
Do you have a specific error message where you need help on?

@cr7pt0gr4ph7 Please do not add manual review comments for things that pre-commit detected already - unless you have a fix in mind, that is non-obvious.

resetreboot committed:
Clean up and linting

resetreboot commented:
I finally went and checked the log for the workflow output and applied fixes to this @JoergAtGithub hopefully it works now. I will also need to check this code against Mixxx itself and check we haven't broken anything. Will do tomorrow.

resetreboot committed:
fix: more linting (pre-commit doesn't work in local, sorry folks!)

m0dB commented:
Thanks for all the hard work on this, @resetreboot ! Unfortunately, I will not be able to look into this or test it before the weekend, but I will do so as soon as possible!

resetreboot committed:
Fix for main deck variable not working

cr7pt0gr4ph7 reviewed (changes requested):
JS file looks good to me now 👍 My main remaining request: Could you reformat the XML file to use a consistent indentation of 4 spaces (instead of a mixture of spaces and tabs)?

@Swiftb0y Pinging you as you seem to be involved with the ESLint config:
- `var DJCi500` is used by old mapping files, is still [recommended by the Mixxx Wiki](https://github.com/mixxxdj/mixxx/wiki/midi%20scripting#script-file-header) but rejected by eslint. It works because a top-level `var DJCi500 = ...` the same as `globalThis.DJCi500 = ...`.
- `const DJCi500` only works if it is followed by `this.DJCi500 = DJCi500` (or `globalThis.DJCi500 = DJCi500`).
- Newer mapping files seem to omit the `var` completely: `DJCi500 = ...` - which is the same as `globalThis.DJCi500 = ...`
- It would also be possible to use:
  ```js
  // eslint-disable-next-line prefer-const
  var DJCi500 = ...
  ```

So... what is the preferred style & can we update either the Wiki or the ESlint config to reflect it?

Commit:
corrected wiki link, sanitized line-ends and tabs

m0dB commented:
Hi @resetreboot , I have created this PR with formatting fixes https://github.com/resetreboot/mixxx/pull/1

The wiki link was broken (two spaces after the tag). I replaced all tabs with spaces and replaced the dos line-ends. I also ran pre-commit on both files.

Locally, I rebased this on latest main and ran ctest. There are 3 tests failing, but these also fail on main itself.

resetreboot committed:
Merge pull request #1 from m0dB/hercules-inpulse-500-mapping-formatting

corrected wiki link, sanitized line-ends and tabs

Swiftb0y commented:
> var DJCi500 is used by old mapping files, is still [recommended by the Mixxx Wiki](https://github.com/mixxxdj/mixxx/wiki/midi%20scripting#script-file-header) but rejected by eslint. It works because a top-level var DJCi500 = ... the same as globalThis.DJCi500 = ....

Can you elaborate why that doesn't work (with the `// eslint-disable`)?

resetreboot committed:
Fix the var stuff and jogWheel channels

resetreboot committed:
Merge branch 'hercules-inpulse-500-mapping' of github.com:resetreboot/mixxx into hercules-inpulse-500-mapping

resetreboot commented:
Hey folks, since part of this code was also meant to use the most modern practices for mappings (like using ComponentJS) I removed the `var` from the definition as it seems it is the recommended practice, and also fixed a small bug for jogWheels.

cr7pt0gr4ph7 commented on res/controllers/Hercules-DJControl-Inpulse-500-script.js:
```diff
@@ -0,0 +1,1528 @@
+    }
+};
+
+DJCi500.shutdown = function() {
+    //cleanup
+    midi.sendShortMsg(0x90, 0x05, 0x00); //turn browser led off
+    midi.sendShortMsg(0xB0, 0x7F, 0x7E);
+
+};
```
```suggestion
DJCi500.shutdown = function() {
    // Cleanup
    midi.sendShortMsg(0x90, 0x05, 0x00); // Turn browser led off
    midi.sendShortMsg(0xB0, 0x7F, 0x7E);
};
```

Swiftb0y commented:
> > > var DJCi500 is used by old mapping files, is still [recommended by the Mixxx Wiki](https://github.com/mixxxdj/mixxx/wiki/midi%20scripting#script-file-header) but rejected by eslint. It works because a top-level var DJCi500 = ... the same as globalThis.DJCi500 = ....
> > 
> > 
> > Can you elaborate why that doesn't work (with the `// eslint-disable`)?
> 
> I'm not sure if I understand your question correctly, so forgive me if this doesn't answer your question: ESLint warns about any usage of `var` but can be prevented from doing so by prefixing the offending line with `// esling-disable-next-line prefer-const`.

Yes, that is the recommended solution is this particular case.

cr7pt0gr4ph7 reviewed (changes requested):
Left a list of documentation nitpicks that I would prefer to see fixed. Otherwise I'm happy with the code. Great work! :+1:

resetreboot committed:
fix: Comments and var with Eslint comment

resetreboot committed:
fix: range and key not working on correct deck and loop pad 8 fix

resetreboot commented:
Yeah, I just made a couple last fixes. It is running correctly on Mixxx 2.5.0 too, @cr7pt0gr4ph7

resetreboot committed:
fix: typo

m0dB commented:
Hi, I finally have been able to test this mapping thoroughly. I found two issues: 1) loop pad 8 not working, but I see you fixed tis in the meantime. 2) the faders have no effect when shift is pressed down. This happened to me when I was pressing shift to use it in combination with the CUE button to just to the start of the track, while moving the fader up at the same time. Now, this happened to me several times so it would be great to fix it. Apart from this: it works very well. Thanks again for all the hard work on this!

resetreboot commented:
Oh, the controller must be sending over a different channel when you are pressing shift and moving the fader. When you mention the faders, you mention _the volume faders_ right? Or some other?

cr7pt0gr4ph7 reviewed (approved):
Code looks good. I can't give any input on the problems brought up by @m0db, though these should definitely be resolved before merging.  I'll be happy to do a final round of review after these issues have been confirmed to be resolved, but I can't be of much help until then.

resetreboot committed:
fix: volume and pitch faders didn't work with shift

resetreboot commented:
No problem @cr7pt0gr4ph7 just pushed (and verified as working) both volume and pitch faders with shift pressed (both weren't working when shift was pressed and the fix for both was the same and quick)

cr7pt0gr4ph7 reviewed (approved):
LGTM with respect to the code itself. Can't test the functionality as I don't have the hardware, so someone else should retest.

m0dB commented:
I will retest tomorrow. It would be great if we can also get this into 2.5.1. Once merged to main, could you create a MR for 2.5 as well? Same for the manual.

m0dB commented:
A minor discrepancy between mapping and manual: in transport mode, slider range reset is pad 5, in the manual it's pad 8. Do you want me to correct the manual? (Including the svg)

m0dB commented:
And another discrepancy: the values for Loop with shift are different than what the manual says.

m0dB commented:
Assuming that in these cases the manual is wrong and the mapping is correct, everything works perfectly! Great job, @resetreboot, very much appreciated! IMO this can be merged but maybe we should first correct the manual? If you do the .rst, I will do the .svgs.

I tried to use the mapping with 2.5 but it doesn't work. I am pretty sure an earlier version did work, so I am not sure if this is expected. Is this something you want to address? It would be great to have this mapping included with 2.5.1!

m0dB commented:
Hm, I just noticed that I get a ton of warnings on startup:

<details><summary>Details</summary>
<p>

warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2397:19 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2398:29 Variable "isArguments" is used before its declaration at 11256:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2399:40 Variable "isBuffer" is used before its declaration at 11402:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2400:51 Variable "isTypedArray" is used before its declaration at 12177:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2522:7 Variable "baseEach" is used before its declaration at 2836:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2648:19 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2655:19 Variable "getTag" is used before its declaration at 6063:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2658:13 Variable "isBuffer" is used before its declaration at 11402:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2683:11 Variable "isSet" is used before its declaration at 12114:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2691:11 Variable "isMap" is used before its declaration at 11791:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2769:14 Variable "setTimeout" is used before its declaration at 6644:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2884:33 Variable "computed" is used before its declaration at 2888:15.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:2886:37 Variable "computed" is used before its declaration at 2888:15.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3072:14 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3311:22 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3312:22 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3313:42 Variable "getTag" is used before its declaration at 6063:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3314:42 Variable "getTag" is used before its declaration at 6063:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3323:24 Variable "isBuffer" is used before its declaration at 11402:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3324:14 Variable "isBuffer" is used before its declaration at 11402:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3332:29 Variable "isTypedArray" is used before its declaration at 12177:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3363:37 Variable "getTag" is used before its declaration at 6063:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3455:37 Variable "getTag" is used before its declaration at 6063:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3487:16 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3668:21 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3669:32 Variable "isBuffer" is used before its declaration at 11402:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3670:44 Variable "isTypedArray" is used before its declaration at 12177:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3674:15 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3692:45 Variable "isArguments" is used before its declaration at 11256:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3694:15 Variable "isArguments" is used before its declaration at 11256:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3862:46 Variable "previous" is used before its declaration at 3863:15.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:3947:14 Variable "setToString" is used before its declaration at 6656:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4209:37 Variable "seen" is used before its declaration at 4210:15.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4248:11 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4281:37 Variable "createSet" is used before its declaration at 5458:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:14158:14 Variable "upperFirst" is used before its declaration at 15190:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:10998:14 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4475:11 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4478:47 Variable "stringToPath" is used before its declaration at 6734:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6922:24 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4814:33 Variable "getSymbols" is used before its declaration at 6030:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4826:33 Variable "getSymbolsIn" is used before its declaration at 6047:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:4839:20 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5104:26 Variable "wrapper" is used before its declaration at 5105:17.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5113:46 Variable "getData" is used before its declaration at 5893:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5130:46 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5420:9 Variable "setData" is used before its declaration at 6634:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5471:19 Variable "getTag" is used before its declaration at 6063:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5527:42 Variable "getData" is used before its declaration at 5893:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5558:41 Variable "setData" is used before its declaration at 6634:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9126:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9171:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5860:14 Variable "setToString" is used before its declaration at 6656:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9340:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9365:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5871:43 Variable "getSymbols" is used before its declaration at 6030:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:5883:45 Variable "getSymbolsIn" is used before its declaration at 6047:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6157:10 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6157:29 Variable "isArguments" is used before its declaration at 11256:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:11483:32 Variable "isTypedArray" is used before its declaration at 12177:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6267:14 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6267:32 Variable "isArguments" is used before its declaration at 11256:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6322:11 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:1858:19 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:1662:35 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9552:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6445:13 Variable "cache" is used before its declaration at 6451:11.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:6446:11 Variable "cache" is used before its declaration at 6451:11.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9589:12 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9593:12 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9677:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9706:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9747:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9766:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9796:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9816:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:9892:18 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:14862:20 Variable "attempt" is used before its declaration at 15245:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:8985:11 Variable "previous" is used before its declaration at 8989:13.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:1420:40 Variable "_" is used before its declaration at 17081:7.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:1420:75 Variable "_" is used before its declaration at 17081:7.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:10651:47 Variable "isArray" is used before its declaration at 11284:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:12836:20 Variable "mergeWith" is used before its declaration at 13470:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:14628:44 Variable "upperFirst" is used before its declaration at 15190:9.
warning [Controller] file:///Users/maarten/git/mixxx/res/controllers/lodash.mixxx.js:16991:30 Variable "chainAll" is used before its declaration at 16998:13.
warning [Controller] Lodash is deprecated and will be removed in a future release of Mixxx; refer to the page https://github.com/mixxxdj/mixxx/wiki/Lodash-Migration for migrations instructions
warning [Controller] ERROR! new Deck() called without specifying any deck numbers
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1003
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1005
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1007
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1009
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1019
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1021
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1023
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1025
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1033
warning [Controller] Too many arguments, ignoring 1
warning [Controller] @file:///Users/maarten/git/mixxx/res/controllers/Hercules-DJControl-Inpulse-500-script.js:1034
warning [Controller] Too many arguments, ignoring 1
warning [Controller] options.deck and option.group are both set; options.deck will take priority
warning [Controller] options.deck and option.group are both set; options.deck will take priority
warning [Controller] options.deck and option.group are both set; options.deck will take priority
warning [Controller] options.deck and option.group are both set; options.deck will take priority
warning [LibraryScanner 1] QSqlDatabasePrivate::removeDatabase: connection 'MIXXX-2' is still in use, all queries will cease to work.


</p>
</details>

resetreboot commented:
That's strange, I'm on 2.5.0 and I've been rocking it no problem. 

About the discrepancies, yeah, I think we need to fix the manual. Trust the code. I will take a look along the week. THank you for the extra testing!

m0dB commented:
Ok, I will try again with 2.5 branch tomorrow. 

The warning are with this branch do. You don’t get them?

resetreboot commented:
I will take a look tomorrow, I have a session coming so they should pop out if they are there. I'm about to head to bed here.

m0dB commented:
And a final note, in the Setting page it still has the 1d postfix in the name, and only list DJ Phatso as the author.

resetreboot commented:
@m0dB I can confirm I have those messages about extra parameters. I will look at the docs of the Mixxx API to see how these callbacks have changed since 2.4 and adjust them so we have 0 warnings. Also, the lodash stuff is this code's fault? Because we aren't loading it anywhere.

resetreboot committed:
fix: warnings at startup, name and authors

resetreboot committed:
fix: remove commented lodash loading

resetreboot commented:
Fixed the warnings and the author data, and version name.

m0dB commented:
I tested on main and 2.5 and all good! Do you want me to create the MR for 2.5? I have a branch ready.

I see there is a eslint warning left. I see there is a comment to disable the check but apparently it doesn't work... No idea why. Can you try

```var DJCi500 = {}; // eslint-disable-line```

?

m0dB commented:
I have created the PR for 2.5 here https://github.com/mixxxdj/mixxx/pull/14491 

In that PR I changed the line
```
var DJCi500 = {};
```
to:
```
var DJCi500 = {}; // eslint-disable-line

```
in order to satisfy pre-commit.

m0dB commented:
I created a PR with final fixes for manual to match the mapping in https://github.com/resetreboot/manual/pull/3

resetreboot committed:
fix: fix linter
